src/command_helpers.rs

```diff
@@ -311,7 +311,19 @@ pub(crate) fn objects_from_files(files: &[Arc<Path>], dst: &Path) -> Result<Vec<
         // Hash the dirname. This should prevent conflicts if we have multiple
         // object files with the same filename in different subfolders.
         let mut hasher = hash_map::DefaultHasher::new();
-        hasher.write(dirname.to_string().as_bytes());
+
+        // Make the dirname relative (if possible) to avoid full system paths influencing the sha
+        // and making the output system-dependent
+        let mut dirname = dirname.to_string();
+
+        // Here we allow using std::env::var (instead of Build::getenv) because
+        // CARGO_* variables always trigger a rebuild when changed
+        #[allow(clippy::disallowed_methods)]
+        if let Ok(root) = std::env::var("CARGO_MANIFEST_DIR") {
```

Comment:
Please use `var_os` and `to_string_lossy` to support non utf-8 characters:
```suggestion
        if let Ok(root) = std::env::var_os("CARGO_MANIFEST_DIR") {
```